### ISSUE TITLE
Smb2 client rename support

### ISSUE THREAD

**Issue by hagaygo** [TalAloni/SMBLibrary#11]
Hi Tal,

SMB1 got SMB1FileStoreHelper class which support rename using SetFileInformation.

It seems SMB2FileStore class doesn't implement SetFileInformation so file rename is not supported on SMB2, am i missing something ?

Thanks in advance for any help/pointers.

**Comment by TalAloni** [comment]
Hi Hagay,
SMB1 has a rename command which is then executed using the FileStore's SetFileInformation method.
SMB2 more closely resemble the FileStore, it has a SetInfo command which uses the FileRenameInformationType2 parameter to rename files.

**Comment by hagaygo** [comment]
Hi again,

Thanks for the answer.

Either i don't understand your answer or i was not clear enough.

For clarification i am talking about SMB client side only.

After implementing the list,read/write operations with your help on on https://github.com/TalAloni/SMBLibrary/issues/9, I am trying to implement a rename file operation (for remote file on remote machine).

In your answer you mentioned using SetInfo command.
There is a SetInfoHelper class under SMBLibrary.Server.SMB2 with a static method called GetSetInfoResponse, but as far i can understand its for server only implementation.

Is there any chance your could provide a code example for SMB client rename operation (SMB1/2) ?

**Comment by TalAloni** [comment]
Sorry for misunderstanding, here is a code example: (please update to the latest revision for it to work properly)

    SMB2Client client = new SMB2Client();
    bool success = client.Connect(System.Net.IPAddress.Parse("192.168.1.121"), SMBTransportType.DirectTCPTransport);
    if (success)
    {
        NTStatus status = client.Login(String.Empty, username, password);
        if (status == NTStatus.STATUS_SUCCESS)
        {
            SMB2FileStore fileStore = client.TreeConnect("Shared", out status) as SMB2FileStore;
            if (fileStore != null)
            {
                object handle;
                FileStatus fileStatus;
                // Open existing file
                status = fileStore.CreateFile(out handle, out fileStatus, "1.txt", AccessMask.GENERIC_ALL, 0, ShareAccess.Read, CreateDisposition.FILE_OPEN, CreateOptions.FILE_NON_DIRECTORY_FILE, null);
                if (status == NTStatus.STATUS_SUCCESS)
                {
                    FileRenameInformationType2 renameInfo = new FileRenameInformationType2();
                    renameInfo.FileName = "2.txt";
                    status = fileStore.SetFileInformation(handle, renameInfo);
                    fileStore.CloseFile(handle);
                }
            }
        }
    }

**Comment by hagaygo** [comment]
Great , working just fine after your latest addition.

For me , AccessMask.GENERIC_ALL returned access denied (for every user i tried), what currently worked o.k for all the users i tried was AccessMask.MAXIMUM_ALLOWED. 

Thanks again for your fast and productive responses. :)
